include/deal.II/sundials/ida.h

```diff
@@ -0,0 +1,653 @@
+//-----------------------------------------------------------
+//
+//    Copyright (C) 2017 by the deal.II authors
+//
+//    This file is part of the deal.II library.
+//
+//    The deal.II library is free software; you can use it, redistribute
+//    it, and/or modify it under the terms of the GNU Lesser General
+//    Public License as published by the Free Software Foundation; either
+//    version 2.1 of the License, or (at your option) any later version.
+//    The full text of the license can be found in the file LICENSE at
+//    the top level of the deal.II distribution.
+//
+//---------------------------------------------------------------
+
+#ifndef dealii_sundials_ida_h
+#define dealii_sundials_ida_h
+
+#include <deal.II/base/config.h>
+#ifdef DEAL_II_WITH_SUNDIALS
+
+#include <deal.II/base/logstream.h>
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/parameter_handler.h>
+#include <deal.II/base/conditional_ostream.h>
+#include <deal.II/base/mpi.h>
+#include <deal.II/lac/vector.h>
+#include <deal.II/lac/vector_view.h>
+#include <deal.II/lac/vector_memory.h>
+
+#include <ida/ida.h>
+#include <ida/ida_spils.h>
+#include <ida/ida_spgmr.h>
+#include <ida/ida_spbcgs.h>
+#include <ida/ida_sptfqmr.h>
+#include <nvector/nvector_serial.h>
+#include <sundials/sundials_math.h>
+#include <sundials/sundials_types.h>
+
+#include <memory>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+// Shorthand notation for IDA error codes.
+#define AssertIDA(code) Assert(code >= 0, ExcIDAError(code))
+
+namespace SUNDIALS
+{
+
+  /**
+   * Interface to SUNDIALS IDA (Implicit Differential-Algebraic) solver.
```

Comment:
maybe introduce abbreviation here:`Implicit Differential-Algebraic (IDA)` and then you can use it below?